geonode/base/views.py

```diff
@@ -55,31 +57,46 @@ def batch_modify(request, ids, model):
     if model == 'Map':
         Resource = Map
     template = 'base/batch_edit.html'
+    ids = request.POST.get("ids")
 
-    if "cancel" in request.POST:
+    if "cancel" in request.POST or not ids:
         return HttpResponseRedirect(
             '/admin/{model}s/{model}/'.format(model=model.lower())
         )
 
     if request.method == 'POST':
         form = BatchEditForm(request.POST)
         if form.is_valid():
-            for resource in Resource.objects.filter(id__in=ids.split(',')):
-                resource.group = form.cleaned_data['group'] or resource.group
-                resource.owner = form.cleaned_data['owner'] or resource.owner
-                resource.category = form.cleaned_data['category'] or resource.category
-                resource.license = form.cleaned_data['license'] or resource.license
-                resource.date = form.cleaned_data['date'] or resource.date
-                resource.language = form.cleaned_data['language'] or resource.language
-                new_region = form.cleaned_data['regions']
-                if new_region:
-                    resource.regions.add(new_region)
-                keywords = form.cleaned_data['keywords']
-                if keywords:
-                    resource.keywords.clear()
-                    for word in keywords.split(','):
-                        resource.keywords.add(word.strip())
-                resource.save(notify=True)
+            keywords = [keyword.strip() for keyword in
+                        form.cleaned_data.pop("keywords").split(',') if keyword]
+            regions = form.cleaned_data.pop("regions")
+            ids = form.cleaned_data.pop("ids")
+            if not form.cleaned_data.get("date"):
+                form.cleaned_data.pop("date")
+
+            resources = Resource.objects.filter(id__in=ids.split(','))
```

Comment:
this may be very inefficient when ids array is very large. I'm not forcing anything here but this may be the way:
https://dba.stackexchange.com/questions/91247/optimizing-a-postgres-query-with-a-large-in